ui/components/app/collectible-details/index.scss

```diff
@@ -53,18 +53,40 @@ $spacer-break-small: 16px;
     overflow-wrap: break-word;
   }
 
-  &__contract-link,
-  &__image-link {
-    word-break: break-all;
+  &__contract-wrapper {
+    max-width: calc(100% - #{$link-title-width});
+  }
+
+  &__contract-copy-button {
+    @include H6;
 
-    @media screen and (max-width: $break-small) {
-      overflow: hidden;
-      text-overflow: ellipsis;
-      white-space: nowrap;
-      width: 90%;
+    width: 80px;
+    display: flex;
+    align-items: flex-start;
+    justify-content: center;
+    background-color: transparent;
+    cursor: pointer;
+    color: var(--ui-4);
+    border: 0;
+
+    &:active {
+      transform: scale(0.97);
     }
   }
 
+  &__contract-link {
+    overflow: hidden;
+    text-overflow: ellipsis;
+    white-space: nowrap;
+  }
+
+  &__image-link {
+    overflow: hidden;
+    text-overflow: ellipsis;
+    white-space: nowrap;
+    max-width: 332px;
```

Comment:
This feels like a "magic number" -- we should consider documenting why we use this width somewhere.